pkg/dataflatten/json.go

```diff
@@ -1,25 +1,49 @@
 package dataflatten
 
 import (
+	"bufio"
+	"bytes"
 	"encoding/json"
 	"fmt"
+	"io"
 	"os"
 )
 
 func JsonFile(file string, opts ...FlattenOpts) ([]Row, error) {
-	rawdata, err := os.ReadFile(file)
+	f, err := os.Open(file)
+	defer f.Close()
 	if err != nil {
 		return nil, err
 	}
-	return Json(rawdata, opts...)
+
+	return Json(bufio.NewReader(f), opts...)
 }
 
-func Json(rawdata []byte, opts ...FlattenOpts) ([]Row, error) {
-	var data interface{}
+func Json(r io.Reader, opts ...FlattenOpts) ([]Row, error) {
+	decoder := json.NewDecoder(r)
+	var objects []interface{}
+
+	for {
+		var object interface{}
+		err := decoder.Decode(&object)
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+
+			return nil, fmt.Errorf("unmarshalling json: %w", err)
+		}
+		objects = append(objects, object)
+	}
 
-	if err := json.Unmarshal(rawdata, &data); err != nil {
-		return nil, fmt.Errorf("unmarshalling json: %w", err)
+	// if we have a single json object, don't flatten as an array
+	if len(objects) == 1 {
+		return Flatten(objects[0], opts...)
 	}
 
-	return Flatten(data, opts...)
+	return Flatten(objects, opts...)
```

Comment:
I'm a little leery of this switch. My gut sense is that it's going to be hard to write SQL that handles it.

Reply:
split out to `kolide_jsonl` table